### PR TITLE
Festival search filters

### PR DESCRIPTION
**Pull request by pldespaigne**
- [x] add `orgName` to the algolia movie index
- [x] create a seller filter linkt to the algolia movie index
- [x] refactor the genres filter to use algolia
- [x] refactor the country filter to use algolia
- [x] refactor the language filter to use algolia
  - [x] refactor language key in the algolia movie index
- [x] refactor the status filter to use algolia
- [x] refactor the budget filter to use algolia
  - [x] create the budget key in the algolia movie index
- [x] add search by keywords
- [x] clean the movie index from useless data
- [ ] add auto-configuration of index to the backend-ops
- [x] prevent empty page at the beginning (display some movies from the store)
- [x] clean the code

### CODE REVIEW

File: apps/festival/festival/src/app/marketplace/title/list/list.component.html [hunk 1]
```diff
@@ -25,13 +25,14 @@ <h1 class="mat-headline">Library</h1>
     <mat-form-field appearance="outline">
       <mat-label>Search for titles, directors, keywords, ...</mat-label>
       <mat-icon matPrefix svgIcon="magnifying_glasses"></mat-icon>
-      <input matInput type="text" [formControl]="searchbarTextControl" autocomplete="off"
-      [matAutocomplete]="searchbar" #autoCompleteInput>
-      <mat-autocomplete #searchbar="matAutocomplete" (optionSelected)="addSearchbarValue($event)">
+      <input matInput type="text" [formControl]="searchbarTextControl">
+      <!-- autocomplete="off"
```
Comment (GrandSchtroumpf) [inline apps/festival/festival/src/app/marketplace/title/list/list.component.html]: Remove totally. We would use the `<algolia-autocomplete>` anyway if we want autocomplete

File: apps/festival/festival/src/app/marketplace/title/list/list.component.html [hunk 2]
```diff
@@ -168,6 +191,9 @@ <h6>Filters</h6>
   <!-- List || card of result -->
   <article>
     <ng-container *ngIf="(movieSearchResults$ | async) as movies">
+      <div fxLayout="row" fxLayoutAlign="end start">
+        <h6>There are {{ movies.length }} movies available!</h6>
```
Comment (GrandSchtroumpf) [inline apps/festival/festival/src/app/marketplace/title/list/list.component.html]: remove div above, use "text-align: right" instead

File: apps/festival/festival/src/app/marketplace/title/list/list.component.ts [hunk 3]
```diff
@@ -118,44 +123,105 @@ export class ListComponent implements OnInit {
...
+    // );
+
+    this.orgSearchResults$ = this.filterForm.seller.valueChanges.pipe(
       debounceTime(200),
       distinctUntilChanged(),
-      startWith(''),
-      pluck('text'),
-      switchMap(text => this.movieIndex.search(text)),
-      pluck('hits'),
+      filter(text => text !== undefined && text !== '' && text !== ' '),
```
Comment (GrandSchtroumpf) [inline apps/festival/festival/src/app/marketplace/title/list/list.component.ts]: why not just `filter(text => !!text?.trim())` ?

**Comment by pldespaigne** [comment]
This PR has gone straight to merge conflict hell.
Closing... you can find the clean version [here](https://github.com/blockframes/blockframes/pull/2294)

**Pull request closed without merging**